lib/observability/tracing/tracing.go

```diff
@@ -124,6 +125,14 @@ func (c *Config) CheckAndSetDefaults() error {
 	return nil
 }
 
+func (c Config) Endpoint() string {
```

Comment:
A value receiver here will copy the entire contents of the config just to extract the endpoint. Not a huge deal, but given that this code is used to measure performance, we might want to take a pointer receiver and avoid the copy.